src/rhsmlib/facts/pkg_arches.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2020 ATIX AG
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+import logging
+import subprocess
+
+from rhsmlib.facts import collector
+from rhsmlib.facts.hwprobe import HardwareCollector
+
+log = logging.getLogger(__name__)
+
+
+class SupportedArchesCollector(collector.FactsCollector):
+    """
+    Class used for collecting packages architectures of a host
+    """
+
+    def __init__(self, arch=None, prefix=None, testing=None):
+        super(SupportedArchesCollector, self).__init__(
+            arch=arch,
+            prefix=prefix,
+            testing=testing)
+
+    def get_arches_on_debuntu(self):
+        """
+        Try to return content of all supported packages architectures
+        :return: dictionary containing architectures
+        Otherwise empty dictionary is returned.
+        """
+        arches = []
+
+        try:
+            output = str(subprocess.check_output("dpkg --print-architecture", shell=True))
```

Comment:
The output of `subprocess.check_output()` is in bytes. You cannot convert it to `str` on Python 3 using `str()` and expect some nice result. You should use `decode()` method. You should also remove the trailing `\n`. Example:

```python
arch = b'amd64\n'
arch = arch.decode('UTF-8')
arch.rstrip('\n')
```

Reply:
Implemented as suggested, in the latest commit.